toolchain/sem_ir/builtin_function_kind.cpp

```diff
@@ -68,6 +68,9 @@ using Bool = BuiltinType<InstId::BuiltinBoolType>;
 // TODO: This only matches i32 for now. Support iN for all N, and the
 // Core.BigInt type we use to implement for integer literals.
 using AnyInt = BuiltinType<InstId::BuiltinIntType>;
+
+// Constraint that requires the type to be the type type.
+using TypeT = BuiltinType<InstId::BuiltinTypeType>;
```

Comment:
The `T` suffix suggests to me that this would be a constraint on a generic parameter, eg`fn F[T:! type](x: T) -> T` would be `auto(TypeT) -> TypeT`, matching how `IntT` behaves. Maybe just `Type`, to match `Bool`?